### PR TITLE
Check NodeState before updating partition table

### PR DESCRIPTION
**Pull request by mdogan**
_Fixes some issues found during hot-restart testing_

### CODE REVIEW

**Comment by metanet** [comment]
:+1:

File: hazelcast/src/main/java/com/hazelcast/partition/impl/PartitionStateOperation.java [hunk 1]
```diff
@@ -46,6 +48,11 @@ public PartitionStateOperation(PartitionRuntimeState partitionState, boolean syn
 
     @Override
     public void run() {
+        NodeEngineImpl nodeEngine = (NodeEngineImpl) getNodeEngine();
+        if (nodeEngine.getNode().getState() == NodeState.PASSIVE) {
+            getLogger().warning("Partition table received before startup is not completed. Caller: " + getCallerAddress());
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/partition/impl/PartitionStateOperation.java]: should 'not' be added in the message? Should it be 'before startup is complete..'

**Comment by pveentjer** [comment]
I guess it will be tricky to add tests for this since it is difficult to force the cluster in a certain state.

**Comment by mdogan** [comment]
@pveentjer: we have tests on enterprise repo, since we can't test hot-restart on OSS. See https://github.com/hazelcast/hazelcast-enterprise/pull/453

**Pull request merged**